sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <chrono>
+#include "opentelemetry/core/timestamp.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/sdk/trace/recordable.h"
+#include "opentelemetry/trace/canonical_code.h"
+#include "opentelemetry/trace/span_id.h"
+#include "opentelemetry/trace/trace_id.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+/**
+ * SpanData is an immutable representation of all data collected by a span.
+ */
+class SpanData final : public Recordable
+{
+public:
+  // The trace id for this span.
+  opentelemetry::trace::TraceId GetTraceId() const noexcept = 0;
```

Comment:
Can you add a .cc file and a span_data_test.cc?

> opentelemetry::trace::TraceId GetTraceId() const noexcept = 0;

Shouldn't compile since "= 0" will only work on virtual functions, so it doesn't look like this file isn't getting checked by the compiler.